File: abc/abc140/abc140_b.cs
// https://atcoder.jp/contests/abc140/submissions/7387602

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        var B = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        var C = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

        var previous = -1;
        var result = 0;
        for (int i = 0; i < N; i++)
        {
            result += B[A[i] - 1];
            if (previous + 1 == A[i])
            {
                result += C[previous - 1];
            }
            previous = A[i];
        }

        Console.WriteLine(result);
    }
}

File: abc/abc140/abc140_b.fs
// https://atcoder.jp/contests/abc140/submissions/12018234

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int
let B = stdin.ReadLine().Split() |> Array.map int
let C = stdin.ReadLine().Split() |> Array.map int

let rec solver p r i =
    if i = N then r
    elif p + 1 = A.[i] then solver A.[i] (r + B.[A.[i] - 1] + C.[p - 1]) (i + 1)
    else solver A.[i] (r + B.[A.[i] - 1]) (i + 1)

solver -1 0 0 |> stdout.WriteLine
